### ISSUE TITLE
A question about Target Length & 

### ISSUE THREAD

**Issue by xiaoxiaoAurora** [OpenNMT/OpenNMT-py#1131]
Hi，I want to implement a translation model that take L1 as a input and the inter-language as the output byt openNMT-py. During training, the accuracy rate is above 99.0，and the validation accuracy can reached  89. However,  After running the tools/multi-bleu.pl ,  the value of BLEU can only reach 43.  
    And，**the translation process** gets the target sentence, but the repetition of the sentence/phrase/word appears in the target sentence and  the target sentence is obviously redundant and too long .  I am very confused about this situation. How can I deal with this situation?

**Comment by bpopeters** [comment]
I'm not sure I can speak about anything particular to your specific case because I'm not clear exactly what your task is, but in general it's important to understand that the validation accuracy metric measures a very different thing from BLEU score. Validation accuracy is how often the model predicts the correct target token, _given the previous correct target sequence_. At translation time, the model does not have access to these correct answers, only to its own previous predictions. If these previous predictions are not very good, then the model can easily predict something that is very far from the reference. Also, at validation time the predicted and gold target sequences are exactly the same length, whereas the target may be any length at translation time.

**Comment by xiaoxiaoAurora** [comment]
Thank you very much for your reply. I think I could have found a new adjustment methods, I will try it. And my task is to generate a Chinese sentence written by a second language learner(possibly with a syntax error) via openNMT with a Chinese sentence written by a native Chinese speaker as input.

**Comment by vince62s** [comment]
what is the dataset size you are using ?

**Comment by xiaoxiaoAurora** [comment]
Training set size is 30000.

**Comment by vince62s** [comment]
it is way too small to learn anything.

**Comment by xiaoxiaoAurora** [comment]
Ok, I will expand the training data. Thank you very much for your patience.

**Comment by bpopeters** [comment]
I wouldn't be so quick to say it's too small, because how much data you need is task-dependent. 30k is nothing for machine translation, but I work on some tasks where that would be quite a lot. I've never worked on your task, but it definitely sounds less data intensive than MT.

Are you really trying to convert L1 Chinese into L2 Chinese? Converting the other way sounds -- i.e. fixing the errors that non-native speakers make -- sounds much more natural to me. And if the L1 version is more or less the same as the L2 sentence but with the errors fixed, then it sounds related to existing work on [grammatical error correction](https://nlpprogress.com/english/grammatical_error_correction.html), where some of the training sets are under 100k sentence pairs.

**Comment by vince62s** [comment]
I think it's really difficult not to overfit with so few data, not talking of having a decent LM fluency for output. For summarization, CNN/DM has 200K examples and we struggle. (well the task is more complicated).
Anyway, if your training accuracy is 99, good chance it overfitted.

**Comment by xiaoxiaoAurora** [comment]
> I wouldn't be so quick to say it's too small, because how much data you need is task-dependent. 30k is nothing for machine translation, but I work on some tasks where that would be quite a lot. I've never worked on your task, but it definitely sounds less data intensive than MT.
> 
> Are you really trying to convert L1 Chinese into L2 Chinese? Converting the other way sounds -- i.e. fixing the errors that non-native speakers make -- sounds much more natural to me. And if the L1 version is more or less the same as the L2 sentence but with the errors fixed, then it sounds related to existing work on [grammatical error correction](https://nlpprogress.com/english/grammatical_error_correction.html), where some of the training sets are under 100k sentence pairs.

I'm really trying to convert L1 Chinese into L2 Chinese, and this is just my first step. Next, I will use this data for further work, may be  L1-L2 parallel dependency parsing comparison.
